packaging/arch/PKGBUILD

```diff
@@ -122,40 +110,66 @@ check() {
   mv $srcdir/xxx-info data/info
 }
 
-package_snapd-git() {
+package() {
+  cd "$pkgname"
   export GOPATH="$srcdir/go"
 
-  # Install snap, snapctl, snap-update-ns, snap-seccomp, snap-exec and snapd
-  # executables
-  install -d -m 755 "$pkgdir/usr/bin/"
-  install -m 755 -t "$pkgdir/usr/bin" \
-          "$GOPATH/bin/snap" \
-          "$GOPATH/bin/snapctl"
-
-  install -d -m 755 "$pkgdir/usr/lib/snapd"
-  install -m 755 -t "$pkgdir/usr/lib/snapd" \
-          "$GOPATH/bin/snap-update-ns" \
-          "$GOPATH/bin/snap-exec" \
-          "$GOPATH/bin/snap-seccomp" \
-          "$GOPATH/bin/snapd"
-
-  # Install snap-confine
-  make -C "$srcdir/$pkgbase/cmd" install DESTDIR="$pkgdir"
-
-  # Install script to export binaries paths of snaps and XDG_DATA_DIRS for their
-  # .desktop files
-  make -C "$srcdir/$pkgbase/data/env" install DESTDIR="$pkgdir"
-
-  # Install D-Bus service files
-  make -C "$srcdir/$pkgbase/data/dbus" install \
-       DBUSSERVICESDIR=/usr/share/dbus-1/services \
-       DESTDIR="$pkgdir"
-
-  # Install systemd units
-  make -C "$srcdir/$pkgbase/data/systemd" install \
-       BINDIR=/bin \
-       SYSTEMDSYSTEMUNITDIR=/usr/lib/systemd/system \
-       DESTDIR="$pkgdir"
+  # Install bash completion
+  install -Dm644 data/completion/snap \
+    "$pkgdir/usr/share/bash-completion/completion/snap"
+  install -Dm644 data/completion/complete.sh \
+    "$pkgdir/usr/lib/snapd/complete.sh"
+  install -Dm644 data/completion/etelpmoc.sh \
+    "$pkgdir/usr/lib/snapd/etelpmoc.sh"
+
+  # Install systemd units, dbus services and a script for environment variables
+  make -C data/ install \
+     DBUSSERVICESDIR=/usr/share/dbus-1/services \
+     BINDIR=/usr/bin \
+     SYSTEMDSYSTEMUNITDIR=/usr/lib/systemd/system \
+     SNAP_MOUNT_DIR=/var/lib/snapd/snap \
+     DESTDIR="$pkgdir"
+
+  # Install polkit policy
+  install -Dm644 data/polkit/io.snapcraft.snapd.policy \
+    "$pkgdir/usr/share/polkit-1/actions/io.snapcraft.snapd.policy"
+
+  # Install executables
+  install -Dm755 "$GOPATH/bin/snap" "$pkgdir/usr/bin/snap"
+  install -Dm755 "$GOPATH/bin/snapctl" "$pkgdir/usr/bin/snapctl"
+  install -Dm755 "$GOPATH/bin/snapd" "$pkgdir/usr/lib/snapd/snapd"
+  install -Dm755 "$GOPATH/bin/snap-seccomp" "$pkgdir/usr/lib/snapd/snap-seccomp"
+  install -Dm755 "$GOPATH/bin/snap-update-ns" "$pkgdir/usr/lib/snapd/snap-update-ns"
+  install -Dm755 "$GOPATH/bin/snap-exec" "$pkgdir/usr/lib/snapd/snap-exec"
+
+  # pre-create directories
+  install -dm755 "$pkgdir/var/lib/snapd/snap"
+  install -dm755 "$pkgdir/var/cache/snapd"
+  install -dm755 "$pkgdir/var/lib/snapd/assertions"
+  install -dm755 "$pkgdir/var/lib/snapd/desktop/applications"
+  install -dm755 "$pkgdir/var/lib/snapd/device"
+  install -dm755 "$pkgdir/var/lib/snapd/hostfs"
+  install -dm755 "$pkgdir/var/lib/snapd/mount"
+  install -dm755 "$pkgdir/var/lib/snapd/seccomp/bpf"
```

Comment:
Nitpick: we should eventually add `/var/lib/snapd/apparmor/{profiles,snap-confine}` here.